packages/vite/src/node/server/environment.ts

```diff
@@ -117,12 +122,33 @@ export class DevEnvironment extends BaseEnvironment {
       this.pluginContainer!.resolveId(url, undefined),
     )
 
-    this.hot = context.hot || createNoopHotChannel()
-
     this._crawlEndFinder = setupOnCrawlEnd()
 
     this._remoteRunnerOptions = context.remoteRunner ?? {}
-    context.remoteRunner?.transport?.register(this)
+
+    this.hot = context.transport
+      ? isWebSocketServer in context.transport
+        ? context.transport
+        : normalizeHotChannel(context.transport)
+      : createNoopHotChannel()
+    this.hot.on('vite:fetchModule', async (data, client, invoke) => {
+      if (!invoke) return
```

Comment:
What is `invoke` here? A silent exit feels wrong here. Why would someone trigger this event intentionally without this? 🤔

Reply:
Changed to an error in f892a8d92bd0856eddb6fb3d303b1ec29cf99ba7 👍